src/Castle.Core/DynamicProxy/Internal/TypeUtil.cs

```diff
@@ -134,6 +135,32 @@ public static Type GetTypeOrNull(object target)
 			return target.GetType();
 		}
 
+		internal static bool HasAnyOverridableDefaultImplementations(this Type interfaceType)
+		{
+			Debug.Assert(interfaceType != null);
+			Debug.Assert(interfaceType.IsInterface);
+
+			var cache = hasAnyOverridableDefaultImplementationsCache;
+			lock (cache)
+			{
+				if (!cache.TryGetValue(interfaceType, out var result))
+				{
+					foreach (var method in interfaceType.GetAllInstanceMethods())
```

Comment:
Note that while this operation (`GetAllInstanceMethods`) is relatively expensive, if we didn't pre-screen interfaces using this new `HasAnyOverridableDefaultImplementations` method, we'd still end up calling it for each interface (in `InterfaceMembersWithDefaultImplementationCollector`'s base class). So we're really just calling it a little earlier, and a later, repeated call then becomes much cheaper since it uses a cache internally.